superset/security/manager.py

```diff
@@ -417,6 +418,27 @@ def can_access_table(self, database: "Database", table: "Table") -> bool:
 
         return True
 
+    def raise_for_dashboard_access(self, dashboard: "Dashboard") -> None:
```

Comment:
It's worth adding feature flag checks here, too, just in case this method was accidentally invoked outside of a feature flag enabled brach.

As a general note, unless it's refactoring that will also benefit the base case, new code for feature flags should all wrap behind the flag because it would also make cleaning up the feature flag easier (hope we don't have to do it for this case!).